airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/AsyncStreamConsumer.java

```diff
@@ -126,7 +126,13 @@ public void accept(final String messageString, final Integer sizeInBytes) throws
    */
   private Optional<PartialAirbyteMessage> deserializeAirbyteMessage(final String messageString) {
     final Optional<PartialAirbyteMessage> messageOptional = Jsons.tryDeserialize(messageString, PartialAirbyteMessage.class)
-        .map(partial -> partial.withSerialized(messageString));
+        .map(partial -> {
+          if (partial.getRecord() != null) {
+            return partial.withSerialized(partial.getRecord().getData().toString());
+          } else {
+            return partial.withSerialized(messageString);
```

Comment:
one thing I realised while writing tests, this is now slightly confusing because in the case where this is a record, this now contains the data itself

in the case where this is a state message, this contains the entire state message